src/mercury/Mercury.ts

```diff
@@ -136,7 +136,19 @@ export default class Mercury implements MercuryInterface {
   }
 
   private notDid(did: Domain.DID | undefined): did is undefined {
-    return !(did instanceof Domain.DID);
+    if (!did) {
+      return true;
+    }
+    if (!did?.method || !did.methodId || !did.schema || !did.toString) {
+      return true;
+    }
+    if (typeof did.method !== "string" || typeof did.methodId !== "string" || typeof did.schema !== "string") {
+      return true;
+    }
+    if (typeof did.toString !== "function") {
+      return true;
+    }
+    return false
```

Comment:
this looks odd, why is all this extra logic needed? shouldn't this all be asserted from a Domain.DID?